src/allmydata/testing/web.py

```diff
@@ -147,7 +138,7 @@ class _FakeTahoeUriHandler(Resource, object):
 
     isLeaf = True
 
-    data = attr.ib(default=attr.Factory(dict))
+    data: Dict[bytes, bytes] = attr.ib(default=attr.Factory(dict))
```

Comment:
Maybe a `allmydata.util.dictutil.BytesKeyDict`? That'll actually enforce it at runtime too.